src/components/Counter/Counter.jsx

```diff
@@ -1,71 +1,102 @@
-import { useEffect, useState } from 'react';
-import styles from './Counter.css';
+import { useEffect, useState, useReducer } from 'react'
 
-const colors = {
-  yellow: 'rgb(236, 222, 153)',
-  green: 'rgb(52, 211, 153)',
-  red: 'rgb(239, 68, 68)',
+const yellow= 'rgb(236, 222, 153)'
+const initialCount = 0
+
+const countReducer = (state, action) => {
+  switch (action.type) {
+    case 'INCREMENT':
+      return state + 1
+
+    case 'DECREMENT':
+      return state - 1
+
+    case 'RESET':
+      return 0
+
+    default:
+      return state;
+  }
 };
 
+const colorReducer = (color, action) => {
+  switch (action.type) {
+    case 'COLOR':
+      return { color: action.payload.color }
+    default:
+      return color;
+  }
+};
 export default function Counter() {
-  const [count, setCount] = useState(0);
-  const [currentColor, setCurrentColor] = useState(colors.yellow);
+  // const [count, setCount] = useState(0)
+  // const [currentColor, setCurrentColor] = useState(pinkRGB)
+
+  const [currentColor, dispatch] = useReducer(colorReducer, yellow)
+  const [count, dispatchCount] = useReducer(countReducer, initialCount)
 
   useEffect(() => {
     if (count === 0) {
-      setCurrentColor(colors.yellow);
+      // setCurrentColor(pinkRGB)
+      dispatch({ type: 'COLOR', payload: { color: (yellow) } })
     }
 
     if (count > 0) {
-      setCurrentColor(colors.green);
+      // setCurrentColor(`rgb(52, 211, 153)`)
+      dispatch({ type: 'COLOR', payload: { color: `rgb(52, 211, 153)` } })
```

Comment:
Here in the dispatch you can pass the color into the payload object, similar to line 40 since you have it declared on line 4.